### PR TITLE
Add a full-screen modal template for password reset prompt

### PR DESCRIPTION
**Pull request by cnasc**
# Description

Another chunk of #2166, this PR introduces the UI by which users can input the new password to re-encrypt their account. I'd like to think of this as just an initial iteration; in particular a next step would be to port over the (slightly) better password validation from account creation. (a next-next step would be to have a better standard password validation for all components to use.)

The component is hooked up to the `FullScreenModals` render path, but nothing yet dispatches an action that can trigger it.

image

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [ ] This PR only contains configuration changes (package.json, etc.)
  - [x] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [x] My code follows the style guidelines of this project, including naming conventions
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] If my code adds or changes components, I have written corresponding stories with Storybook
- [x] I have performed a self-review of my own code
- [x] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: unlock-app/src/stories/interface/FullScreenModals.stories.js [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
   WalletCheck,
   PasswordPrompt,
+  ResetPasswordPrompt,
 } from '../../components/interface/modal-templates'
 import { KindOfModal } from '../../unlockTypes'
 import doNothing from '../../utils/doNothing'
 
+const emailState = {
+  userDetails: {
+    email: 'geoff@bitconnect.gov',
```
Comment (julien51) [inline unlock-app/src/stories/interface/FullScreenModals.stories.js]: Who's this guy????

Comment (julien51) [reply unlock-app/src/stories/interface/FullScreenModals.stories.js]: I keep seeing him so now I have to ask!

**Comment by cnasc** [comment]
> Did @smombartz provide the design? I am surprised the button look different than usually?

I don't think we've had any pro-work on the modal designs. These I kept from the password prompt modal as seen here: https://github.com/unlock-protocol/unlock/pull/3141

They are shared components, so swapping them out for something else should be easy

File: unlock-app/src/stories/interface/FullScreenModals.stories.js (hunk 1 above)
Comment (cnasc) [reply unlock-app/src/stories/interface/FullScreenModals.stories.js]: A fictional character from the Department of Bitconnect!

**Pull request merged**